lib/archethic/account/mem_tables/token_ledger.ex

```diff
@@ -56,59 +45,154 @@ defmodule Archethic.Account.MemTables.TokenLedger do
   """
   @spec add_unspent_output(
           recipient_address :: binary(),
-          utxo :: VersionedUnspentOutput.t()
+          utxo :: VersionedUnspentOutput.t(),
+          from_init? :: boolean()
         ) :: :ok
   def add_unspent_output(
         to_address,
-        %VersionedUnspentOutput{
+        input = %VersionedUnspentOutput{
           unspent_output: %UnspentOutput{
             from: from_address,
             amount: amount,
-            type: {:token, token_address, token_id},
-            timestamp: %DateTime{} = timestamp
-          },
-          protocol_version: protocol_version
-        }
+            type: {:token, token_address, token_id}
+          }
+        },
+        from_init? \\ false
       )
       when is_binary(to_address) and is_binary(from_address) and is_integer(amount) and amount > 0 and
              is_binary(token_address) and is_integer(token_id) and token_id >= 0 do
-    spent? =
-      case :ets.lookup(@unspent_output_index_table, to_address) do
-        [] ->
-          false
+    case :ets.lookup(@unspent_output_index_table, to_address) do
+      [{^to_address, true}] ->
+        append_input_to_db(to_address, input, from_init?)
```

Comment:
I don't think this will work, as the input writer opens files in exclusive mode. This means, if the file already exists, it will not write to the file anymore.

Reply:
if you compile the node with  `ARCHETHIC_INPUTS_THRESHOLD=3`  you will see by doing some integration the inputs aren't collected.

Reply:
I have updated the code, and tried with threshold 3